contacts/views.py

```diff
@@ -1,5 +1,33 @@
-from django.http import HttpResponse
+from django.shortcuts import render
+from django.views import generic
+from .models import Contact, Address
 
 
-def index(request):
-    return HttpResponse("Sample View")
+class ContactView(generic.ListView):
+    template_name = 'contacts/contact.html'
+    context_object_name = 'my_contacts'
+
+    def get_queryset(self):
+        """
+        Return contacts by name
+        """
+        return Contact.objects.order_by('name')
+
+
+# class DetailView(generic.DetailView):
+#     model = Contact
+#     template_name = 'contacts/detail.html'
+#
+#     def get_queryset(self):
+#         """
+#         Return details of contact
+#         """
+#         return Contact.objects.get(id=Contact.id)
+
+
+def detail(request, contact_id):
+    contact_details = Contact.objects.get(id=contact_id)
+    address_details = Address.objects.get(id=contact_id)
```

Comment:
the query to `Address` model will fetch wrong results, think on the mistake you're doing here! :P

Reply:
done